### ISSUE TITLE
Update the Synapse Client version requirement

### ISSUE THREAD

**Issue by Aryllen** [Sage-Bionetworks/dccvalidator#504]
https://github.com/Sage-Bionetworks/dccvalidator/blob/bfc42ee702d43a9ab2629f63df405993e7a99743/DESCRIPTION#L77

Should be latest, 2.4.0.

**Comment by kelshmo** [comment]
This is a plain text field and not automatically checked. Update the [installation instructions](https://github.com/Sage-Bionetworks/dccvalidator/blob/master/vignettes/deploying-dccvalidator.Rmd#L34-L52) and add a [py requirements file](https://pip.pypa.io/en/stable/user_guide/#requirements-files). 

I'm not exactly sure where the py requirements file should live as the installation should work both is a user clones the repo and if the user installs the package. For a package install, I have in other packages added files to the inst directory so that it gets copied into the top-level package directory. Then it gets copied in the users working directory with some custom code.

**Comment by kelshmo** [comment]
Or add assert statement to the app. Example: 
```
synapseclient <- reticulate::import("synapseclient")
syn_version <- synapseclient["__version__"]
stopifnot(compareVersion(syn_version, "2.3.0") >= 0)
```

**Comment by danlu1** [comment]
Solved by PR[#526](https://github.com/Sage-Bionetworks/dccvalidator/pull/526).
